### PR TITLE
`Development`: Remove deprecated router module in client tests

### PR DESCRIPTION
**Pull request by JohannesWt**
#### General

- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/development-process.html#naming-conventions-for-github-pull-requests).

#### Client
- [x] I **strictly** followed the [client coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client/).

### Motivation and Context

Currently a lot of client tests use the `RouterTestingModule` which is deprecated.

### Description

This PR replaces the `RouterTestingModule` with `provideRouter()` or `RouterModule.forRoot()` according to [Angular Documentation](https://v17.angular.io/api/router/testing/RouterTestingModule)

### Steps for Testing

All client tests must pass

[](https://artemis-test1.artemis.cit.tum.de)
[](https://artemis-test2.artemis.cit.tum.de)
[](https://artemis-test3.artemis.cit.tum.de)
[](https://artemis-test4.artemis.cit.tum.de)
[](https://artemis-test5.artemis.cit.tum.de)
[](https://artemis-test6.artemis.cit.tum.de)
[](https://artemis-test9.artemis.cit.tum.de)

#### Code Review
- [x] Code Review 1
- [x] Code Review 2

### Test Coverage

unchanged

## Summary by CodeRabbit

- **Refactor**
	- Updated test setups across multiple components to replace `RouterTestingModule` with `RouterModule.forRoot()` or `provideRouter`, enhancing routing configuration in tests.
	- Streamlined import statements for improved clarity and maintainability.
- **Chores**
	- Consolidated imports and updated providers in various test files to reflect the new routing setup.

### CODE REVIEW

File: src/test/javascript/spec/component/footer/footer.component.spec.ts [hunk 1]
```diff
@@ -1,19 +1,20 @@
...
 
 describe('FooterComponent', () => {
     let component: FooterComponent;
     let fixture: ComponentFixture<FooterComponent>;
 
     beforeEach(async () => {
         await TestBed.configureTestingModule({
-            declarations: [FooterComponent, MockPipe(ArtemisTranslatePipe)],
-            imports: [ArtemisTestModule, RouterTestingModule, TranslateModule.forRoot()],
+            declarations: [FooterComponent, MockPipe(ArtemisTranslatePipe), RouterModule.forRoot([])],
```
Comment (pzdr7) [inline src/test/javascript/spec/component/footer/footer.component.spec.ts:15]: Declaration or import?

File: src/test/javascript/spec/component/tutorial-groups/shared/tutorial-group-detail.component.spec.ts [hunk 2]
```diff
@@ -1,5 +1,5 @@
 import { TutorialGroupDetailComponent } from 'app/course/tutorial-groups/shared/tutorial-group-detail/tutorial-group-detail.component';
-import { RouterTestingModule } from '@angular/router/testing';
+import {} from '@angular/router/testing';
```
Comment (pzdr7) [inline src/test/javascript/spec/component/tutorial-groups/shared/tutorial-group-detail.component.spec.ts:2]: Could we remove this empty import?

File: src/test/javascript/spec/component/file-upload-exercise/file-upload-exercise-detail.component.spec.ts [hunk 3]
```diff
@@ -1,13 +1,13 @@
...
+import { ActivatedRoute, provideRouter } from '@angular/router';
 import { HttpHeaders, HttpResponse } from '@angular/common/http';
 import { of } from 'rxjs';
 import { ArtemisTestModule } from '../../test.module';
 import { FileUploadExerciseDetailComponent } from 'app/exercises/file-upload/manage/file-upload-exercise-detail.component';
 import { MockFileUploadExerciseService, fileUploadExercise } from '../../helpers/mocks/service/mock-file-upload-exercise.service';
 import { JhiLanguageHelper } from 'app/core/language/language.helper';
 import { AlertService } from 'app/core/util/alert.service';
-import { RouterTestingModule } from '@angular/router/testing';
+import {} from '@angular/router/testing';
```
Comment (pzdr7) [inline src/test/javascript/spec/component/file-upload-exercise/file-upload-exercise-detail.component.spec.ts:10]: Empty import

File: src/test/javascript/spec/component/modeling-submission/modeling-submission.component.spec.ts [hunk 4]
```diff
@@ -8,8 +8,8 @@ import { MockSyncStorage } from '../../helpers/mocks/service/mock-sync-storage.s
 import { MockParticipationWebsocketService } from '../../helpers/mocks/service/mock-participation-websocket.service';
 import { LocalStorageService, SessionStorageService } from 'ngx-webstorage';
 import { TranslateService } from '@ngx-translate/core';
-import { RouterTestingModule } from '@angular/router/testing';
-import { ActivatedRoute } from '@angular/router';
+import {} from '@angular/router/testing';
```
Comment (pzdr7) [inline src/test/javascript/spec/component/modeling-submission/modeling-submission.component.spec.ts:11]: Empty import

File: src/test/javascript/spec/component/programming-assessment/code-editor-tutor-assessment-container.component.spec.ts [hunk 5]
```diff
@@ -32,12 +32,12 @@ import { Course } from 'app/entities/course.model';
...
 import { ProgrammingSubmissionService } from 'app/exercises/programming/participate/programming-submission.service';
 import { ComplaintResponse } from 'app/entities/complaint-response.model';
-import { ActivatedRoute, Router, convertToParamMap } from '@angular/router';
+import { ActivatedRoute, Router, convertToParamMap, provideRouter } from '@angular/router';
 import { ProgrammingExerciseService } from 'app/exercises/programming/manage/services/programming-exercise.service';
 import { CodeEditorRepositoryFileService } from 'app/exercises/programming/shared/code-editor/service/code-editor-repository.service';
 import { CodeEditorFileBrowserComponent } from 'app/exercises/programming/shared/code-editor/file-browser/code-editor-file-browser.component';
 import { FileType } from 'app/exercises/programming/shared/code-editor/model/code-editor.model';
-import { RouterTestingModule } from '@angular/router/testing';
+import {} from '@angular/router/testing';
```
Comment (pzdr7) [inline src/test/javascript/spec/component/programming-assessment/code-editor-tutor-assessment-container.component.spec.ts:40]: Empty import

File: src/test/javascript/spec/component/text-submission-assessment/text-submission-assessment.component.spec.ts [hunk 6]
```diff
@@ -26,7 +26,7 @@ import { TextBlock, TextBlockType } from 'app/entities/text/text-block.model';
 import { Feedback, FeedbackType } from 'app/entities/feedback.model';
 import { ComplaintResponse } from 'app/entities/complaint-response.model';
 import { AlertService } from 'app/core/util/alert.service';
-import { RouterTestingModule } from '@angular/router/testing';
+import {} from '@angular/router/testing';
```
Comment (pzdr7) [inline src/test/javascript/spec/component/text-submission-assessment/text-submission-assessment.component.spec.ts:29]: Empty import

File: src/test/javascript/spec/component/tutorial-groups/course-tutorial-groups/course-tutorial-group-card.component.spec.ts [hunk 7]
```diff
@@ -1,5 +1,5 @@
 import { ComponentFixture, TestBed } from '@angular/core/testing';
-import { RouterTestingModule } from '@angular/router/testing';
+import {} from '@angular/router/testing';
```
Comment (pzdr7) [inline src/test/javascript/spec/component/tutorial-groups/course-tutorial-groups/course-tutorial-group-card.component.spec.ts:2]: Empty import

**Review by pzdr7 — APPROVED** [review]
Code

**Review by JohannesStoehr — DISMISSED** [review]
Refactoring looks good to me

**Review by pzdr7 — APPROVED** [review]
Reapprove

**Pull request merged**